commit 64aeb1209c42c4897113af20e24b0a4dc3519add
Author: Evgeni Golov <evgeni@grml.org>
Date:   Mon Mar 11 09:17:58 2013 +0100

    Fixing condition for setting rootfs_dest_backing.
    
    The fix for the opt_link check in d5ed942 broke the code as it inverted the
    meaning. Previously it ran the code iff opt_link was empty, as [ -n ] is
    true. In the case opt_link was not empty, the code evaluated to [ -n foo],
    which is invalid syntax and thus false. This, however, was almost the expected
    behaviour, as rootfs_dest_backing is needed for opt_union and (in read-only
    mode) for opt_link.

diff --git a/scripts/boot/9990-misc-helpers.sh b/scripts/boot/9990-misc-helpers.sh
index cb3940b..fb7c644 100755
--- a/scripts/boot/9990-misc-helpers.sh
+++ b/scripts/boot/9990-misc-helpers.sh
@@ -1482,7 +1482,7 @@ activate_custom_mounts ()
 		# ignore the loop below and set rootfs_dest_backing=$dest
 		local rootfs_dest_backing
 		rootfs_dest_backing=""
-		if [ -n "${opt_link}" ]
+		if [ -n "${opt_link}" ] || [ -n "${opt_union}" ]
 		then
 			for d in /live/rootfs/*
 			do
